### PR TITLE
Fix bug 237 and 271

### PR DESCRIPTION
**Pull request by nilangekarss**
This is a fix for #271 #237.
As ScheduleName is a mandatory parameter to be passed for creating a snapshot schedule, this PR will make sure that error is thrown if other snapschedule parameters are passed and not scheduleName.

This PR restrict the use of retentionHours and expirationHours to be set for the additional snapshot which is getting created. Earlier expirationHours were not allowed with the base snapshots but now both retentionHours and expirationHours are not allowed to be associated with a base snapshot while creating a snapshot schedule.

### CODE REVIEW

**Comment by nilangekarss** [comment]
@wdurairaj please review the code

File: hpedockerplugin/hpe_storage_api.py [hunk 1]
```diff
@@ -345,6 +345,15 @@ def volumedriver_create(self, name, opts=None):
...
                                                          opts)
+                schedule_opts = valid_snap_schedule_opts[1:]
+                for s_o in schedule_opts:
+                    if s_o in input_list:
+                        if "scheduleName" not in input_list:
+                            msg = (_('scheduleName is a mandatory parameter'
+                                     ' for creating a snapshot schedule'))
+                            LOG.error(msg)
+                            response = json.dumps({u"Err": msg})
+                            return response
```
Comment (sonivi) [inline hpedockerplugin/hpe_storage_api.py]: can we optimise as below:

```
for s_o in schedule_opts:
    if s_o in input_list and "scheduleName" not in input_list:
        msg = (_('scheduleName is a mandatory parameter for creating a snapshot schedule'))
        LOG.error(msg)
        response = json.dumps({u"Err": msg})
        return response
   break
```

Comment (nilangekarss) [reply hpedockerplugin/hpe_storage_api.py]: Done

**Comment by nilangekarss** [comment]
@sonivi @wdurairaj Please review the code and merge if okay.
Added UT for snapshot schedule.
Addressed Vivek's comments.

**Pull request merged**